replica/database.cc

```diff
@@ -1013,31 +1013,46 @@ void database::remove(const table& cf) noexcept {
     }
 }
 
-future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_name, timestamp_func tsf, bool snapshot) {
-    auto& ks = find_keyspace(ks_name);
-    auto uuid = find_uuid(ks_name, cf_name);
-    lw_shared_ptr<table> cf;
-    try {
-        cf = _column_families.at(uuid);
-        drop_repair_history_map_for_table(uuid);
-    } catch (std::out_of_range&) {
-        on_internal_error(dblog, fmt::format("drop_column_family {}.{}: UUID={} not found", ks_name, cf_name, uuid));
-    }
-    dblog.debug("Dropping {}.{}", ks_name, cf_name);
-    remove(*cf);
-    cf->clear_views();
-    co_await cf->await_pending_ops();
-    co_await _querier_cache.evict_all_for_table(cf->schema()->id());
-    auto f = co_await coroutine::as_future(truncate(ks, *cf, std::move(tsf), snapshot));
-    co_await cf->stop();
-    f.get(); // re-throw exception from truncate() if any
+future<> database::drop_column_family(table& cf) {
+    auto uuid = cf.schema()->id();
+    drop_repair_history_map_for_table(uuid);
+    remove(cf);
+    cf.clear_views();
+    co_await cf.await_pending_ops();
+    co_await _querier_cache.evict_all_for_table(uuid);
+}
+
+future<std::vector<foreign_ptr<lw_shared_ptr<table>>>> database::get_table_on_all_shards(sharded<database>& sharded_db, utils::UUID uuid) {
+    std::vector<foreign_ptr<lw_shared_ptr<table>>> table_shards;
```

Comment:
By the way, we could also abstract access to local table, with something like this:

```
class global_table_ptr {
    std::vector<foreign_ptr<lw_shared_ptr<table>>> _table_per_shard;
public:
    static global_table_ptr make_global_table_ptr(sharded<database>& sharded_db, utils::UUID uuid);
    
    table& operator->() const {
        return *_table_per_shard[this_shard_id()];
    }
};
    
```

maybe it's overkill. I like the approach you used too.

Reply:
as I said above, I think that the longer term approach should be in the direction of `sharded<keyspace>` and `sharded<table>` objects.  Not sure how to stitch them to the database shards (maybe define a table::impl and have table contain a lw_shared_ptr<table::impl>).

Reply:
If database itself is sharded, then theoretically table and keyspace are just members of database and don't have to be sharded. Cross-shard operations requiriring a coordinator should go through database which is itself sharded.

Reply:
But with an eye on tablets, the idea is to decouple table objects from database objects, which has a single instance per shard and won't move.